apps/src/templates/teacherDashboard/SectionsPage.jsx

```diff
@@ -89,6 +97,12 @@ class SectionsPage extends Component {
       sections = response;
       onAsyncLoad();
     });
+
+    // If we have a default courseId and/or scriptId, we want to start with our
+    // dialog open. Add a new section with this course/script as default
+    if (defaultCourseId || defaultScriptId) {
+      this.addSection();
+    }
```

Comment:
General flow is we get defaultCourseId/defaultScriptId as query params on sections.js. It passes these in to SectionsPage. If provided, then onMount (after loading some async data) we call `addSection`. In the old flow, this results in the row being edited, in the new flow it launches our dialog. In both cases, we provide the redux store with the course (and also script in new flow).